nxdrive/manager.py

```diff
@@ -134,27 +135,17 @@ def __init__(self, home: Path, /) -> None:
         if Options.force_locale is not None:
             self.set_config("locale", Options.force_locale)
 
-        # Handle synchronization state early
-        sync_enabled = self.dao.get_config("synchronization_enabled")
-        if sync_enabled is not None:
-            # If the option has been set from another conf source, it has the precedence
-            # over the old (and maybe obsolete) value stored in the database
-            if (
-                Options.options["synchronization_enabled"][1] != "default"
-                and Options.synchronization_enabled != sync_enabled
-            ):
-                # Update the value stored in the database using the the global options value
-                self.dao.update_config(
-                    "synchronization_enabled", Options.synchronization_enabled
-                )
-            else:
-                # Update global options using the value stored in the database
-                Options.synchronization_enabled = sync_enabled != "0"
+        # Backward-compatibility: handle synchronization state early
+        if version_le(__version__, "5.1.2"):
+            sync_enabled = self.dao.get_config("synchronization_enabled")
+            if sync_enabled is not None:
+                # Note: no need to handle the case where the sync is disabled because it is the default behavior
+                if sync_enabled != "0":
+                    self.set_feature_state("synchronization", True)
+                self.dao.delete_config("synchronization_enabled")
 
-        if not Options.synchronization_enabled:
-            log.info(
-                ">>> Synchronization features are disabled, only Direct Edit and Direct Transfer will work."
-            )
+        if not Feature.synchronization:
+            log.info(">>> Synchronization features are disabled.")
```

Comment:
I am wondering if this could be moved to a callback? Or if we need it at all?

Maybe the log is not necessary, because we do not log such lines for other features. Moreover, when the feature is disabled later, there is no log.

Reply:
I think that this is not needed anymore too.